protofsm/state_machine.go

```diff
@@ -184,6 +184,10 @@ type StateMachineCfg[Event any, Env Environment] struct {
 	// MsgMapper is an optional message mapper that can be used to map
 	// normal wire messages into FSM events.
 	MsgMapper fn.Option[MsgMapper[Event]]
+
+	// CustomPollInterval is an optional custom poll interval that can be
+	// used to set a quicker interval for tests.
+	CustomPollInterval fn.Option[time.Duration]
```

Comment:
Maybe ditch the option here and then set the custom poll interval to `pollInterval` in the constructor so we don't have to deal with the extra branching.

Reply:
This was useful for unit tests, discussed offline.